pkg/scheduler/objects/application_state_test.go

```diff
@@ -19,15 +19,22 @@
 package objects
 
 import (
+	"fmt"
 	"testing"
 	"time"
 
 	"gotest.tools/v3/assert"
 
 	"github.com/apache/yunikorn-core/pkg/common"
+	"github.com/apache/yunikorn-core/pkg/events"
+	"github.com/apache/yunikorn-scheduler-interface/lib/go/si"
 )
 
 func TestAcceptStateTransition(t *testing.T) {
+	events.CreateAndSetEventSystem()
```

Comment:
It's too much change of the existing tests. 
I suggest that you create a new test case which triggers all state transitions then you verify the events. That results in a more compact and likely smaller change.

Reply:
Done. Thanks.